Content.Server/GameObjects/Components/Weapon/Ranged/Projectile/SharedProjectileWeaponComponent.cs

```diff
@@ -0,0 +1,95 @@
+﻿using Content.Server.GameObjects.Components.Mobs;
+using Content.Server.GameObjects.Components.Projectiles;
+using Content.Server.GameObjects.Components.Sound;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Interfaces.Random;
+using Robust.Shared.IoC;
+using Robust.Shared.Map;
+using Robust.Shared.Maths;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+using System.Collections.Generic;
+
+namespace Content.Server.GameObjects.Components.Weapon.Ranged.Projectile
+{
+    /// <summary>
+    ///     Methods to shoot projectiles.
+    /// </summary>
+    public abstract class SharedProjectileWeaponComponent : Component
```

Comment:
Make this something like `BaseProjectileWeaponComponent` instead. `Shared` is usually used to denote code in the shared project that also exists on the client.